domain/src/test/java/org/oppia/android/domain/topic/TopicControllerTest.kt

```diff
@@ -431,19 +430,11 @@ class TopicControllerTest {
   @Test
   fun testRetrieveChapter_invalidChapter_returnsFailure() {
     topicController.retrieveChapter(
-      INVALID_TOPIC_ID_1, INVALID_STORY_ID_1, INVALID_EXPLORATION_ID_1
-    ).observeForever(mockChapterSummaryObserver)
+      FRACTIONS_TOPIC_ID, FRACTIONS_STORY_ID_0, RATIOS_EXPLORATION_ID_0
+    ).toLiveData().observeForever(mockChapterSummaryObserver)
     testCoroutineDispatchers.runCurrent()
 
     verifyRetrieveChapterFailed()
-    assertThat(chapterSummaryResultCaptor.value.getErrorOrNull()).isInstanceOf(
-      TopicController.ChapterNotFoundException::class.java
-    )
-    assertThat(chapterSummaryResultCaptor.value.getErrorOrNull()).hasMessageThat()
-      .isEqualTo(
-        "Chapter for exploration $INVALID_EXPLORATION_ID_1 not found in story " +
-          "$INVALID_STORY_ID_1 and topic $INVALID_TOPIC_ID_1"
-      )
```

Comment:
It's probably still worth verifying that AsyncResult includes the correct failure.

Reply:
updated the test to verify the error on failure.